frontend/src/app/main/ui/workspace/sidebar/options/menus/typography.cljs

```diff
@@ -102,7 +102,9 @@
         input        (mf/use-ref)
 
         fonts        (mf/use-memo (mf/deps @state) #(filter-fonts @state @fonts/fonts))
-        recent-fonts (mf/deref refs/workspace-recent-fonts)
+        fontsdb      (mf/deref fonts/fontsdb)
+        ;; Filtering deleted fonts
+        recent-fonts (into [] (filter #(some? (get fontsdb (:id %)))) (mf/deref refs/workspace-recent-fonts))
```

Comment:
My suggestion here is that you leave the line that was there before, and add a new line with the filters.

Reply:
Or something like this:
![image](https://github.com/penpot/penpot/assets/843689/ca7c191a-fd83-4de5-9d7b-12082d32f1fa)